erigon-lib/state/aggregator.go

```diff
@@ -1635,7 +1637,7 @@ func (ac *AggregatorRoTx) nastyFileRead(name kv.Domain, from, to uint64) (*seg.R
 type AggregatorRoTx struct {
 	a   *Aggregator
 	d   [kv.DomainLen]*DomainRoTx
-	iis [kv.StandaloneIdxLen]*InvertedIndexRoTx
+	iis map[kv.InvertedIdx]*InvertedIndexRoTx
```

Comment:
A bit sad about replacing array by map (it adding to hot-patch +1 pointer deref and +1 hashing), but probably it’s inavidable cost of flexibility.

Reply:
code is not a bottleneck in our case: disk is (data gormat on disk, indices used, data-locality,  etc…). But there are many maps: IBS has several, SharedDomains too, aggRoTx too

Reply:
- agreed, same feelings with array to map. 
- I think same thing will happen with domain and appendables. 
- I think the direction in which erigon is going -- there might be many components, each registering these datastructures...so we don't know the total number of them. We have to use strings to avoid conflicts.

Reply:
I thin each component will have own aggregator - same as each component has own on DB. They just pass tables config when opening db.